tests/FetchAuthTokenTest.php

```diff
@@ -53,10 +54,20 @@ class_implements($fetcherClass)
         )) {
             $mockFetcher->getQuotaProject()->shouldBeCalledTimes(1);
         }
-        $mockFetcher->fetchAuthToken(Argument::any())
-            ->shouldBeCalledTimes(1)
-            ->will($httpHandler);
+
+        if (is_a($fetcherClass, UpdateMetadataInterface::class, true)) {
```

Comment:
We don't really use `is_a` anymore... you can just use `instanceof` for consistency
```suggestion
        if ($fetcherClass instanceof UpdateMetadataInterface::class) {
```

Reply:
`instanceof` isn't allowing a comparison between string representations : (. For eg, 

```php
class Alpha {}
$str = Alpha::class;
echo ($str instanceof Alpha ? 'true', 'false') . PHP_EOL;

// Prints: false
```

and 
```php
class Alpha {}
$str = Alpha::class;
$a =  ($str instanceof Alpha::class ? 'true', 'false')
```

throws a syntax error `PHP Parse error:  syntax error, unexpected 'class' (T_CLASS), expecting variable (T_VARIABLE) or '$'`

Reply:
I didn't realize that we were dealing with a string and not an object so, you're right, `is_a` is the way to go.